docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,240 @@ Note that:
+  * the plugin is the one that associates the reported config-type list from the external to it's cloud device twin
+  * these config-list files per child-device is also managed by Configuration Management feature from C8Y
+* Consuming/Providing files from/to thin-edge via network
+  * any configuration file exchange to/from the external child device is handled by an HTTP based filetransfer service provided by thin-edge
+
+## Managing Supported Configuration List of child devices 
+
+The external device sends an MQTT message to `tedge/meta/plugin/configuration/<childid>` to announce it's configuration management capability to thin-edge. That MQTT message contains all configurations the external device provides. Thereby each configuration appears with a `type` and with an optional field `path`.
+
+The MQTT message is as below:
+
+```json
+{
+   "configurations": [
+     {
+       "type": "<config type 1>",
+       "path": "</path/to/file/file 1>"
+     },
+     {
+       "type": "<config type 2>",
+       "path": "</path/to/file/file 2>"
+     },
+     ...
+   ]
+}
+```
+
+Example:
+```json
+{
+   "configurations": [
+     {
+       "type": "foo.conf",
+       "path": "/etc/child1/foo.conf"
+     },
+     {
+       "type": "bar.conf",
+       "path": "/etc/child1/bar.conf"
+     }
+   ]
+}
+```
+
+If path is specified, the `c8y_configuration_plugin` consumes/provides the configuration-file from/to the given local filesystem path. If the field `path` is not given, the `c8y_configuration_plugin` make use of the HTTP filetransfer feature of the `tedge_agent` to consume/provide the configuration-file (see [section below](#details-to-aspect-2-filetransfer-fromto-external-device) for more details about HTTP filetransfer). The first case is intended for local processes (running on the thin-edge device) that represent a child-device, and the latter case is intended for external devices.
+
+Whenever the `c8y_configuration_plugin` receivces that MQTT message it stores all contained information to a child-specific TOML file in `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml`. When the file already exists it will be replaced with the new content. The format of that TOML file is according to section [Configuration](#configuration) above. Each individual child's TOML file is stored in a subfolder, where the subfolder name is the `childid`.
+
+Example:
+
+```bash
+    $ tree /etc/tedge/c8y/
+    /etc/tedge/c8y/
+    ├── c8y-configuration-plugin.toml
+    ├── child1
+    │   └── c8y-configuration-plugin.toml
+    └── child2
+        └── c8y-configuration-plugin.toml    
+```
+
+The `c8y_configuration_plugin` adds per child-device the file `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml` implicitely to the child-device's config listed. So the list can always be configured from the cloud. The `type` for this self configuration file is `c8y-configuration-plugin`.
```

Comment:
There's one additional enhancement possible here that will give more control to the child device on this config list file content. If the child device wants to keep the config list in some other format other than toml, let the device declare this special config type `c8y-configuration-plugin` in the bootstrap message that it sends. The device can then map this type to any file with the list of supported config types in any format of its choice. If the child declares this special type in its bootstrap message, we'll skip the creation of this `c8y-configuration-plugin.toml` file on the device and let the device manage it on its own.

Reply:
I will put that to a future extension list.